# Parameters for task sequencer
# Remember that there might be other important constants that can only be changed from the src code.

full_grasper:
  # Name of the arm
  arm_name: panda_arm
  # Name of the hand
  hand_name: right_hand
  # Arm position controller
  arm_pos_controller: position_joint_trajectory_controller
  # Arm velocity controller
  arm_vel_controller: cartesian_velocity_controller
  # Hand position controller
  hand_pos_controller: joint_trajectory_controller
  # Hand velocity controller
  hand_vel_controller: velocity_controller

  # The joint configuration for home
  home_joints: [-0.035, -0.109, -0.048, -1.888, 0.075, 1.797, -0.110] 
  # Transform from object frame to grasp pose frame (xyzrpy) (DEFAULT)
  grasp_transform: [0.0, 0.0, 0.090, -3.1415, 0.0, 0.0]
  # Transform from grasp pose frame to pre-grasp pose frame (xyzrpy)
  pre_grasp_transform: [0.0, 0.0, -0.1, 0.0, 0.0, 0.0]
  # The joint configuration for handover
  handover_joints: [-0.101, 0.161, 0.159, -1.651, 2.023, 2.419, -0.006]
  # The threshold on tau_ext for handover hand opening
  handover_thresh: 1.0

  # The personalized grasp pose maps for different objects
  poses_map:
    "kettle" : [0.066, 0.012, 0.203, -2.973, 0.620, -0.032]
    "teddy" : [0.046, -0.238, 0.110, 3.131, -0.052, -0.400]
    "ball_thumb" : [-0.055, 0.113, 0.07, -3.104, 0.122, -0.241]
    "ball_middle" : [-0.055, 0.113, 0.07, -3.104, 0.122, -0.241]
    "two_cubes" : [-0.055, 0.113, 0.061, -3.104, 0.122, -0.241]
    "bottle" : [0.051, -0.076, 0.077, -2.560, -0.240, 0.353]
    "helmet" : [0.120, -0.039, 0.179, 3.007, 0.200, 0.247]
    "debug" : [0.013, 0.012, 0.338, -1.512, 0.021, -0.005]